### ISSUE TITLE
Ignore Certain Exceptions while Debugging

### ISSUE THREAD

**Issue by MasonEdgel** [microsoft/vscode-cpptools#2374]
Type: Debugger

- OS and Version: Windows 10 Enterprise x64
- VS Code Version: 1.25.1
- C/C++ Extension Version: 0.17.7
- Other extensions you installed (and if the issue persists after disabling them): N/A

There currently doesn't appear to be a way to ignore/skip breaking on specific exceptions when debugging. 

In my specific use case, an Antivirus DLL is injected (beyond my control) that monitors and regulates memory access, and causes a _large_ number of access violation errors in OS libraries.
![log](https://user-images.githubusercontent.com/38443320/43851492-15538454-9af8-11e8-92fd-5df49d9d3063.png)

These errors don't seem to cause functional issues with the program, and I can step over them to continue, but there are often 10, 20, or more steps that cause violations _per line_ in the code I write, and it very quickly becomes unreasonable.

This is with the "All Exceptions" box unchecked, when checked the errors are not logged to the debug console, but execution is still paused and so I am presented with the same problem.
![stack](https://user-images.githubusercontent.com/38443320/43851764-d8c21194-9af8-11e8-830d-2ea85a192acc.png)

Being able to even truly ignore all exceptions and only stop on breakpoints (as mentioned, execution is till paused on thrown exceptions with the "All Exceptions" breakpoint checkbox left clear) would be beneficial, but the ability to specifically ignore types of exceptions would be preferential.

Apologies if this is an existing feature or if an issue was already opened, I spent a good couple of hours digging for a solution and checking to make sure it wasn't already a requested feature and didn't see anything, but I'm new to issue reporting and may have overlooked something.

**Comment by pieandcakes** [comment]
@MasonEdgel Are you debugging Visual C++ or GNU C++? I think our limitation is with VS Code and it not allowing us to exclude specific exceptions. The debugger is probably stopping because of unhandled exception that is happening, 

I'd know better if you can include a [log](https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/FAQs.md#how-to-enable-logging) of you debugging this scenario.

**Comment by MasonEdgel** [comment]
Sorry, debugging with Visual C++.  The exception is occurring outside of code I've written - it happens even with the most basic "Hello World".  I believe this is the relevant log information-
```
Exception thrown at 0x00007FFBF57E324C (ntdll.dll) in diamilo_win32_console.exe: 0xC0000005: Access violation writing location 0x0000000000000000.
<- (E) {"seq":116,"type":"event","event":"stopped","body":{"reason":"exception","threadId":11328,"text":"Unhandled exception at 0x00007FFBF57E324C (ntdll.dll) in program.exe: 0xC0000005: Access violation writing location 0x0000000000000000.","allThreadsStopped":true,"source":{"name":"","path":"","sources":[],"checksums":[]},"line":0,"column":0}}
```

The only other information logged is PDB load failures and stack trace requests, which are all appear to be normal.

Here's a [full log](https://github.com/Microsoft/vscode-cpptools/files/2271516/log.txt) of launching the program and stepping through the error a handful of times, if you want it anyway.

As I mentioned though, I'm certain the exception is caused by the AV's memory management DLL being injected; I've run and debugged the same program on another machine without the AV just fine using the same configuration.  Under this bizarre set of circumstances (and I could see some others), having the ability to skip breaking on these errors would help immensely.

**Comment by pieandcakes** [comment]
Currently, since it is an unhandled exception, they are always caught as we're assuming a program crash shortly after. In this case, my assumption is that the AV program catches it through a different process (which we can't detect) and it doesn't crash. I'll see if I can find more information and I'll let you know.

**Comment by pieandcakes** [comment]
On the other hand if you want the error not to show up in the output window, you can add:
```
"logging": {
     "exceptions": false
}
``` 
to your `launch.json` and it won't fill up your output window, but it will also suppress all other exceptions. You will still stop at them though.

**Comment by francisoliverlee** [comment]
same prblem i meet.
mark one breakpoint,  almost never break cause of too many other exceptions make vscode paused, and a lot of step over, contine i would do....😂😂😂

**Comment by PhilLCar** [comment]
Yep, in the Debug Console `-exec handle SIGUSR1 nostop` or the signal you want to ignore.
![handle signals](https://user-images.githubusercontent.com/16630738/158087627-82f6bfe6-3f4a-48a3-8c6f-2c7622e167d2.png)

**Comment by Challanger524** [comment]
By now it is possible to filter out via **Filter** text-box in the top-right
![image](https://github.com/microsoft/vscode-cpptools/assets/47363301/64057222-ba22-4fa2-a68e-9c8697bd2996)
Using myself: MSYS2 / mingw64 / Clang / `-fsanitize=address`
So in my case filter is: `!libclang_rt.asan_dynamic-x86_64.dll,!ntdll.dll` (`!Loaded` also handy)

### minor issue
Text navigation via `Home` `End` buttons not working despite cursor is active inside the _filter-box_
